perception/autoware_tensorrt_bevdet/src/ros_utils.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 AutoCore, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// cspell:ignore bevdet, RGBHWC, BGRCHW
+
+#include "autoware/tensorrt_bevdet/ros_utils.hpp"
+
+#include <autoware_perception_msgs/msg/detected_object_kinematics.hpp>
+#include <autoware_perception_msgs/msg/object_classification.hpp>
+#include <autoware_perception_msgs/msg/shape.hpp>
+
+#include <preprocess.h>
+
+#include <string>
+#include <vector>
+
+namespace autoware::tensorrt_bevdet
+{
+using Label = autoware_perception_msgs::msg::ObjectClassification;
+
+uint8_t getSemanticType(const std::string & class_name)
+{
+  if (class_name == "car") {
+    return Label::CAR;
+  } else if (class_name == "truck") {
+    return Label::TRUCK;
+  } else if (class_name == "bus") {
+    return Label::BUS;
+  } else if (class_name == "trailer") {
+    return Label::TRAILER;
+  } else if (class_name == "bicycle") {
+    return Label::BICYCLE;
+  } else if (class_name == "motorcycle") {
+    return Label::MOTORCYCLE;
+  } else if (class_name == "pedestrian") {
+    return Label::PEDESTRIAN;
+  } else {
+    return Label::UNKNOWN;
+  }
+}
+
+void box3DToDetectedObjects(
+  const std::vector<Box> & boxes, autoware_perception_msgs::msg::DetectedObjects & bevdet_objects,
+  const std::vector<std::string> & class_names, float score_thre, const bool has_twist = true)
+{
+  for (auto b : boxes) {
+    if (b.score < score_thre) continue;
+    autoware_perception_msgs::msg::DetectedObject obj;
+
+    Eigen::Vector3f center(b.x, b.y, b.z + b.h / 2.);
```

Comment:
I think we don't need center. It is only used for
```
p.x = center.x();
p.y = center.y();
p.z = center.z();
```
we can replace the initialization with the values for center.
we can add a comment indicating that this is the center.